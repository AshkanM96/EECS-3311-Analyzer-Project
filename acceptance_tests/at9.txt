imply
less_than
add
integer_constant(2)
integer_constant(-1)
integer_constant(1)
conjoin
boolean_constant(false)
equals
evaluate
type_check
integer_constant (5)
negative
negative
negative
negative
negative
integer_constant (-5)
type_check
evaluate
reset
reset
reset
evaluate
reset
type_check
reset
integer_constant(1)
evaluate
type_check
reset
integer_constant(1)
add
imply
less_than
conjoin
equals
evaluate
negative
reset
