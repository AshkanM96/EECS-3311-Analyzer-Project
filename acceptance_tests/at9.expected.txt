  Expression currently specified: ?
  Report: Expression is initialized.
->imply
  Expression currently specified: (? => nil)
  Report: OK.
->less_than
  Expression currently specified: ((? < nil) => nil)
  Report: OK.
->add
  Expression currently specified: (((? + nil) < nil) => nil)
  Report: OK.
->integer_constant(2)
  Expression currently specified: (((2 + ?) < nil) => nil)
  Report: OK.
->integer_constant(-1)
  Expression currently specified: (((2 + -1) < ?) => nil)
  Report: OK.
->integer_constant(1)
  Expression currently specified: (((2 + -1) < 1) => ?)
  Report: OK.
->conjoin
  Expression currently specified: (((2 + -1) < 1) => (? && nil))
  Report: OK.
->boolean_constant(False)
  Expression currently specified: (((2 + -1) < 1) => (False && ?))
  Report: OK.
->equals
  Expression currently specified: (((2 + -1) < 1) => (False && (? = nil)))
  Report: OK.
->evaluate
  Expression currently specified: (((2 + -1) < 1) => (False && (? = nil)))
  Report: Error (Expression is not yet fully specified).
->type_check
  Expression currently specified: (((2 + -1) < 1) => (False && (? = nil)))
  Report: Error (Expression is not yet fully specified).
->integer_constant(5)
  Expression currently specified: (((2 + -1) < 1) => (False && (5 = ?)))
  Report: OK.
->negative
  Expression currently specified: (((2 + -1) < 1) => (False && (5 = (- ?))))
  Report: OK.
->negative
  Expression currently specified: (((2 + -1) < 1) => (False && (5 = (- (- ?)))))
  Report: OK.
->negative
  Expression currently specified: (((2 + -1) < 1) => (False && (5 = (- (- (- ?))))))
  Report: OK.
->negative
  Expression currently specified: (((2 + -1) < 1) => (False && (5 = (- (- (- (- ?)))))))
  Report: OK.
->negative
  Expression currently specified: (((2 + -1) < 1) => (False && (5 = (- (- (- (- (- ?))))))))
  Report: OK.
->integer_constant(-5)
  Expression currently specified: (((2 + -1) < 1) => (False && (5 = (- (- (- (- (- -5))))))))
  Report: OK.
->type_check
  Expression currently specified: (((2 + -1) < 1) => (False && (5 = (- (- (- (- (- -5))))))))
  Report: (((2 + -1) < 1) => (False && (5 = (- (- (- (- (- -5)))))))) is type-correct.
->evaluate
  Expression currently specified: (((2 + -1) < 1) => (False && (5 = (- (- (- (- (- -5))))))))
  Report: True
->reset
  Expression currently specified: ?
  Report: OK.
->reset
  Expression currently specified: ?
  Report: Error (Initial expression cannot be reset).
->reset
  Expression currently specified: ?
  Report: Error (Initial expression cannot be reset).
->evaluate
  Expression currently specified: ?
  Report: Error (Expression is not yet fully specified).
->reset
  Expression currently specified: ?
  Report: Error (Initial expression cannot be reset).
->type_check
  Expression currently specified: ?
  Report: Error (Expression is not yet fully specified).
->reset
  Expression currently specified: ?
  Report: Error (Initial expression cannot be reset).
->integer_constant(1)
  Expression currently specified: 1
  Report: OK.
->evaluate
  Expression currently specified: 1
  Report: 1
->type_check
  Expression currently specified: 1
  Report: 1 is type-correct.
->reset
  Expression currently specified: ?
  Report: OK.
->integer_constant(1)
  Expression currently specified: 1
  Report: OK.
->add
  Expression currently specified: 1
  Report: Error (Expression is already fully specified).
->imply
  Expression currently specified: 1
  Report: Error (Expression is already fully specified).
->less_than
  Expression currently specified: 1
  Report: Error (Expression is already fully specified).
->conjoin
  Expression currently specified: 1
  Report: Error (Expression is already fully specified).
->equals
  Expression currently specified: 1
  Report: Error (Expression is already fully specified).
->evaluate
  Expression currently specified: 1
  Report: 1
->negative
  Expression currently specified: 1
  Report: Error (Expression is already fully specified).
->reset
  Expression currently specified: ?
  Report: OK.
